src/txmempool.cpp

```diff
@@ -481,13 +481,16 @@ void CTxMemPool::removeUnchecked(txiter it, MemPoolRemovalReason reason)
     // We increment mempool sequence value no matter removal reason
     // even if not directly reported below.
     uint64_t mempool_sequence = GetAndIncrementSequence();
-
+    const uint256 hash = it->GetTx().GetHash();
     if (reason != MemPoolRemovalReason::BLOCK) {
         // Notify clients that a transaction has been removed from the mempool
         // for any reason except being included in a block. Clients interested
         // in transactions included in blocks can subscribe to the BlockConnected
         // notification.
         GetMainSignals().TransactionRemovedFromMempool(it->GetSharedTx(), reason, mempool_sequence);
+        if (minerPolicyEstimator) {
+            minerPolicyEstimator->removeTx(hash, false);
+        }
```

Comment:
4f22c0da092e1c2287e4bc2d84e646d56abbe133
Why move this into the `reason != BLOCK` case?

Reply:
Any transaction which reason is `BLOCK` will be captured by `MempoolBlockConnect`

Reply:
Ok, I think this should be its own commit with this explanation as it is not relevant to refactoring `processBlock`